### PR TITLE
Migrate Analytics

### PR DESCRIPTION
**Pull request by biyeun**
@orangejenny cc @calellowitz 

please also merge:

https://github.com/dimagi/commcarehq-prelogin/pull/146
https://github.com/dimagi/Vellum/pull/929

### CODE REVIEW

**Comment by orangejenny** [comment]
There's a commcare-hq-deploy change attached to https://github.com/dimagi/commcare-hq/pull/18615/commits/a192e6ccf6afe4092484debd0903c69f63471c66

**Comment by orangejenny** [comment]
Test failures looks real:

`TemplateDoesNotExist: hqwebapp/includes/analytics_all.html`

and

`Warning: Error: The module 'hqwebapp/js/initial_page_data' has already been defined elsewhere.` (need to remove the initial_page_data.js script tag from app manager's mocha.html)

File: .eslintrc.js
Comment (orangejenny) [inline .eslintrc.js]: So nice to see something getting removed from this list.

File: corehq/apps/analytics/static/analytix/js/kissmetrix.js
Comment (orangejenny) [inline corehq/apps/analytics/static/analytix/js/kissmetrix.js]: It looks like `getAbTests` only returns objects with keys for `slug` and `context`, why does this if/else test if ab is an object and look for ab.name and ab.version, and then potentially treat ab as a string? Could use a comment explaining why this is all needed.

**Comment by orangejenny** [comment]
Looks good, just some minor questions/comments.

**Comment by orangejenny** [comment]
@biyeun Added a commit that should fix the js test. The python test will be fixed once the prelogin PR is merged and the submodule update is committed to this PR.

File: corehq/apps/analytics/static/analytix/js/kissmetrix.js
Comment (biyeun) [reply corehq/apps/analytics/static/analytix/js/kissmetrix.js]: i think this is me overengineering 😳

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
Vellum PR is merged, but I'm having trouble running `vellum-to-hq`, therefore holding off on merging this.

**Comment by orangejenny** [comment]
Ran `vellum-to-hq` so https://github.com/dimagi/Vellum/pull/929 is now merged into HQ (https://github.com/dimagi/commcare-hq/commit/194bd316d28a19ddb822f021e06023f1d6b5d497).

**Comment by orangejenny** [comment]
Hey, I didn't notice during review, but the module names here don't match the file names: modules are `analytics/js/foo` but files are `analytix/js/foo.js`. The `analytix` workaround is weird (necessary, but weird), but I'd rather have the module names say `analytix` so we can keep the convention that module names always match filenames - long-term requirejs plan is to to use the variant of `define` where you don't specify a module name at all. Thoughts?

**Pull request merged**
